crates/matrix-sdk/tests/integration/room/joined.rs

```diff
@@ -713,3 +715,51 @@ async fn test_call_notifications_notify_for_rooms() {
 
     room.send_call_notification_if_needed().await.unwrap();
 }
+
+#[async_test]
+async fn test_start_live_location_share_for_room() {
+    let (client, server) = logged_in_client_with_server().await;
+
+    Mock::given(method("PUT"))
+        .and(path_regex(r"^/_matrix/client/r0/rooms/.*/state/org.matrix.msc3672.beacon_info/.*"))
+        .and(header("authorization", "Bearer 1234"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(&*test_json::EVENT_ID))
+        .mount(&server)
+        .await;
+
+    mock_sync(&server, &*test_json::SYNC, None).await;
+
+    let sync_settings = SyncSettings::new().timeout(Duration::from_millis(3000));
+
+    let _response = client.sync_once(sync_settings).await.unwrap();
+
+    let room = client.get_room(&DEFAULT_TEST_ROOM_ID).unwrap();
+
+    let response = room.start_live_location_share(3000, None).await.unwrap();
+
+    assert_eq!(event_id!("$h29iv0s8:example.com"), response.event_id);
+}
+
+#[async_test]
+async fn test_stop_sharing_live_location() {
+    let (client, server) = logged_in_client_with_server().await;
+
+    Mock::given(method("PUT"))
+        .and(path_regex(r"^/_matrix/client/r0/rooms/.*/state/org.matrix.msc3672.beacon_info/.*"))
+        .and(header("authorization", "Bearer 1234"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(&*test_json::EVENT_ID))
+        .mount(&server)
+        .await;
+
+    mock_sync(&server, &*LIVE_LOCATION_SHARING_SYNC, None).await;
```

Comment:
Is it worth having all this hard-coded JSON? Maybe just adding the beacon_info as we do in the previous test and then stopping it would be adequate.

Reply:
+1 to this. If we could inline the JSON here too, that'd be great.

Reply:
After calling `room.start_live_location()`, how do I resync so that the state store has the `beacon_info` I need to test? I'm looking for guidance on setting up the `mock_sync` without using `&*LIVE_LOCATION_SHARING_SYNC`. This will help me get the other tests in correct shape as well. Any examples would be very helpful! Below is simple example of what im trying to achieve

```
...

  mock_sync(&server, &*test_json::SYNC, None).await;
  
   let start_response = room
        .start_live_location_share(3000, None)
        .await
        .expect("Starting live location share should succeed");
    assert_eq!(event_id!("$h29iv0s8:example.com"), start_response.event_id);
    
    // Re-sync so that the start beacon_info is in the state store
    mock_sync()
    assert_eq("Beacon info is the same")
    
     let stop_response =
        room.stop_live_location_share().await.expect("Stopping live location share should succeed");
    assert_eq!(event_id!("$h29iv0s8:example.com"), stop_response.event_id);
    
    // Re-sync so that the stop beacon_info is in the state_store
     mock_sync()
     assert_eq("Beacon info is the same"
```

Reply:
> Could I combine these into 1 test (`start_and_stop_live_location_share` or do you prefer they are separate?

I will always recommend separate tests, one for each thing you are testing. It makes it much easier to understand the test, and, critically for me, it makes it easier to see what has gone wrong if a test fails in future.

Feel free to share e.g. test setup code by putting it in a function or even a `TestSetup` struct if needed.

Reply:
Updated with comprehensive testing, I can take a second pass to try a TestSetup if desired